push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/WebpushPushNotificationSender.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * JBoss, Home of Professional Open Source
+ * Copyright Red Hat, Inc., and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.aerogear.unifiedpush.message.sender;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.fluent.Request;
+import org.apache.http.util.EntityUtils;
+import org.jboss.aerogear.unifiedpush.api.Variant;
+import org.jboss.aerogear.unifiedpush.api.VariantType;
+import org.jboss.aerogear.unifiedpush.message.UnifiedPushMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+
+@SenderType(VariantType.WEBPUSH)
+public class WebpushPushNotificationSender implements PushNotificationSender {
+    
+    private final Logger logger = LoggerFactory.getLogger(WebpushPushNotificationSender.class);
+    
+    private static final String MPS_URL = "https://updates.push.services.mozilla.com/wpush/v1/";
+
+    @Override
+    public void sendPushMessage(Variant variant, Collection<String> clientIdentifiers, UnifiedPushMessage pushMessage,
```

Comment:
We can not determine the browser just from clientIdentifier, consequently we can not decide to which push notification provider we must send a push message (FCM or MPS).

We can use a hack and store the whole URL as a device token, like:
* https://gcm-http.googleapis.com/gcm/send/eIH9WwwyLzE:APA91bGTG1Nun97GHDs4afVq3kV0DWsERqf1_vm7hVf2EGnIbvv3UInCuCcqy4RIL7ZxyVOxE3g3_-V6G3RruRfvoNzNWql_YmrBywRUREwPoVaNxGJ-5sO1LhhnRKMzWPUIysdYg0Rh
* https://updates.push.services.mozilla.com/wpush/v1/gAAAAABXlQW2uvaJJk3Q6hey1cj2PjZYtGaDcY82DffVUF1OiV4Eu6SA1lds8jzKgZCR9JjIbioyv5jKwZQo2n6UxT8yRU3Es1qM2Fxmdv-p0cqGBhh4CjT5QNzlBAFRJ0OTLvisXB8e

Yes, we can determine appropriate push notification provider, but we also have to know "auth secret" and "public key" for each Installation to encrypt a payload of push message. So, I think it would be better to refactor PushNotificationSender to acquire `Collection<Installation> clients` instead of `Collection<String> clientIdentifiers`.

WDYT?